envs/env_4inrow.py

```diff
@@ -70,8 +70,9 @@ def available_moves(self, player):
 
     def is_terminal_state(self):
         status = True
-        if any(not cell for cell in self.board[0]) \
-                or not self.player_status(self.player1):
+        if any(cell == 0 for cell in self.board[0]) \
```

Comment:
This line says the following:  
* If there is an empty cell - return false
* if there isn't an empty cell - check:  
    - if one of the players has a non zero status - return true
    - else - return false

Now, if a player has won/lost, it's indeed a terminal state, no arguing there.  
However, there **could** be an empty cell and the state would still be terminal.  You should rewrite your logic, this is why your tests are failing.